proposals/0000-even-more-improved-integers.md

```diff
@@ -0,0 +1,1107 @@
+# Protocol-oriented integers (take 2)
+
+* Proposal: [SE-NNNN](0000-even-more-improved-integers.md)
+* Authors: [Dave Abrahams](https://github.com/dabrahams), [Maxim Moiseev](https://github.com/moiseev)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Bug: [SR-3196](https://bugs.swift.org/browse/SR-3196)
+* Previous Proposal: [SE-0104][se104]
+
+## Introduction
+
+This proposal is an evolution of [SE-0104][se104]. The goal is still to clean up
+Swifts integer APIs and make them more useful for generic programming.
+
+The language has evolved in ways that affect integers APIs since the time the
+original proposal was approved for Swift 3. We also attempted to implement
+the proposed model in the standard library and found that some essential APIs
+were missing, whereas others could be safely removed.
+
+Major changes to the APIs introduced by this proposal (as compared to
+[SE-0104][se104]) are listed in a [dedicated section](#whats-new-since-se-0104).
+
+## Motivation
+
+Swift's integer protocols don't currently provide a suitable basis for generic
+programming. See [this blog post](http://blog.krzyzanowskim.com/2015/03/01/swift_madness_of_generic_integer/)
+for an example of an attempt to implement a generic algorithm over integers.
+
+The way the `Number` protocol is defined, it does not generalize to floating
```

Comment:
I think this should be `Arithmetic` instead of `Number`, because it refers to the existing Swift 3 protocol.